src/cpp/client.cpp

```diff
@@ -345,6 +345,82 @@ void Client::put_tensor(const std::string& name,
     _report_reply_errors(reply, "put_tensor failed");
 }
 
+// Put bytes into the database
+void Client::put_bytes(const std::string& name,
+                       const void* bytes,
+                       const size_t n_bytes)
+{
+    // Track calls to this API function
+    LOG_API_FUNCTION();
+
+    std::string key = _build_bytes_key(name, false);
+
+    // Send the tensor
+    CommandReply reply = _redis_server->put_bytes(key, bytes, n_bytes);
+
+    if (reply.has_error())
+        throw SRRuntimeException("put_bytes failed");
+}
+
+// Get byte data and return to the user via modifying the
+// supplied void pointer.
+void Client::get_bytes(const std::string& name,
+                       void*& data,
+                       size_t& n_bytes)
+{
+    std::string get_key = _build_bytes_key(name, true);
+    CommandReply reply = _redis_server->get_bytes(get_key);
+
+    if (reply.has_error())
+        throw SRRuntimeException("put_bytes failed");
+
+    // TODO We don't have a way with CommandReply to transfer ownership of a str() reply
+    // to an outside pointer.  For now we do a naive memcopy, 
+    // but we really shouldn't have to do that
+    // We could set the internal str reply value to NULL and the redisreply destructor
+    // which calls free on it will likely do a no-op, but this isn't proven.
+    n_bytes = reply.str_len();
+    data = malloc(n_bytes);
+    std::memcpy(data, (void*)(reply.str()), n_bytes);
+}
+
+// Get byte data and fill an already allocated array 
+// memory space that has the specified size.
+void Client::unpack_bytes(const std::string& name,
```

Comment:
Currently this requires that the buffer and the reply from Redis are exactly the same size. This might be a little overly restrictive especially because the size of the byte string may not be known a priori. I'd suggest that we change this:
```
void Client::unpack_bytes(const std::string& name,
                          void* data,
                          const size_t n_buffer_size,
                          size_t* obj_size
)
```

The check internally should then be that the buffer size is `>=` to the message reply